src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -181,50 +193,60 @@ internal class DocumentAttributes : IChecksummedObject, IObjectWritable
                 IEnumerable<string> folders = null,
                 Optional<SourceCodeKind> sourceCodeKind = default(Optional<SourceCodeKind>),
                 Optional<string> filePath = default(Optional<string>),
+                Optional<string> projectFilePath = default(Optional<string>),
                 Optional<bool> isGenerated = default(Optional<bool>))
             {
                 var newId = id ?? Id;
                 var newName = name ?? Name;
                 var newFolders = folders ?? Folders;
                 var newSourceCodeKind = sourceCodeKind.HasValue ? sourceCodeKind.Value : SourceCodeKind;
                 var newFilePath = filePath.HasValue ? filePath.Value : FilePath;
+                var newProjectFilePath = projectFilePath.HasValue ? projectFilePath.Value : ProjectFilePath;
                 var newIsGenerated = isGenerated.HasValue ? isGenerated.Value : IsGenerated;
 
                 if (newId == Id &&
                     newName == Name &&
                     newFolders == Folders &&
                     newSourceCodeKind == SourceCodeKind &&
                     newFilePath == FilePath &&
+                    newProjectFilePath == ProjectFilePath &&
                     newIsGenerated == IsGenerated)
                 {
                     return this;
                 }
 
-                return new DocumentAttributes(newId, newName, newFolders, newSourceCodeKind, newFilePath, newIsGenerated);
+                return new DocumentAttributes(newId, newName, newFolders, newSourceCodeKind, newFilePath, newProjectFilePath, newIsGenerated);
             }
 
             public void WriteTo(ObjectWriter writer)
             {
+                // these information is volatile. it can be different
+                // per session or not content based value. basically not
+                // persistable. these information will not be included in checksum
                 Id.WriteTo(writer);
 
+                writer.WriteString(FilePath);
```

Comment:
I don't actually understand why DocumentInfo needs ProjectFilePath.  Can you explain why that's necessary for persistable checksums?